tools/src/main/scala/scala/scalanative/build/TargetTriple.scala

```diff
@@ -0,0 +1,453 @@
+// ported from LLVM 887d6ab dated 2023-04-16
+
+//===--- Triple.cpp - Target triple helper class --------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+package scala.scalanative
+package build
+
+import java.nio.ByteOrder
+
+private[scalanative] final case class TargetTriple(
+    arch: String,
+    vendor: String,
+    os: String,
+    env: String
+) {
```

Comment:
I omitted sub-architecture at this time, since parsing it seems complicated, using it also seems complicated, and I wasn't sure what the usecase will be anyway.